ironfish/src/blockchain/blockchain.ts

```diff
@@ -1315,11 +1315,27 @@ export class Blockchain {
 export class VerifyError extends Error {
   reason: VerificationResultReason
   score: number
+  explanation: string | undefined
 
-  constructor(reason: VerificationResultReason, score = 0) {
+  constructor(reason: VerificationResultReason, score = 0, explanation?: string) {
```

Comment:
I think it's a little redundant adding an explanation parameter when we already have a `reason`. I think a better solution would be to add some documentation of what each reason means

Reply:
I added it as an **optional** parameter that can be used in the case of having some additional information that can be attached to the error.

Here is an example usecase:
![image](https://user-images.githubusercontent.com/92462002/184460054-69c827e9-a87c-421c-9cf8-0c4aa4f65d36.png)